### DISCUSSION TITLE
Cannot read properties of undefined (reading 'publish')"

### DISCUSSION THREAD

**Discussion by bobwatcherx** [apollographql/apollo-server#6309] [General]
my code : https://github.com/bobwatcherx/grapapollomain.git
codesanbox : https://codesandbox.io/s/practical-goldwasser-nyy9rw?file=/index.js
PREVIEW APOLLO:https://nyy9rw.sse.codesandbox.io

I'm having problems when I subscribe
on apollo

the error is Cannot read properties of undefined (reading 'publish')"

and when I see an error in the terminal like this
`npx diagnose-endpoint@1.1.0 --endpoint=ws://localhost:4000/graphql
`
**Diagnostics ws://localhost:4000/graphql
️ Connection refused (ECONNREFUSED)
Is the address correct?
Is the server running?**

I've followed various tutorials on the internet, including codesanbox from Apollo itself but it doesn't work

if you can give me a simple code example, thanks

**Comment by glasser** [comment]
This suggests that `pubsub` in `pubsub.publish` is undefined. That's because you are trying to make it refer to the top-level `const pubsub = ` but the line `    hello:async (_,payload,{pubsub})=>{` is definining a function that unpacks its third argument's `pubsub` field into the local variable `pubsub`. Deleting `,{pubsub}` here and elsewhere may help.
